app/models/deployment.rb

```diff
@@ -0,0 +1,30 @@
+class Deployment < ApplicationRecord
+  validates :name,
+    presence: true,
+    uniqueness: { case_sensitive: false },
+    format: { with: NAME_FORMAT },
+    length: { minimum: 1, maximum: 255 }
+
+  belongs_to :cluster
+
+  def container_names
+    (0..count-1).map{ |i| "#{name}-%02d" % (i+1) }
+  end
+
+  def managed_containers
+    Container.exists.
+      where("hostname ~* ?", managed_containers_hostname_regexp).
+      where(cluster: cluster)
+  end
+
+  def wanted_existing_containers
```

Comment:
the name is not intuitive, what's this method for?

Reply:
`wanted` means containers that should be managed by the deployment which is determined by the replica count. For example:
- Deployment **hello** with replica count 2 has **hello-01** and **hello-02** as its wanted containers.
- Deployment **dery** with replica count 3 has **dery-01**, **dery-02**, and **dery-03** as its wanted containers.

`existing_containers` means containers that we have in Pathfinder. So, if we have:
- Containers in Pathfinder: **dery-01** and **dery-04**.
- Wanted containers: **dery-01** and **dery-02**.

`wanted_existing_containers` will be **dery-01** only.

Reply:
if you have suggestions for naming this method, tell me. I'll try to find a better name for this.

Reply:
can we use desired replica and available replica for the terminology in this case

Reply:
Noted.